utils/zerovec/src/varzerovec/components.rs

```diff
@@ -0,0 +1,243 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use super::*;
+use std::marker::PhantomData;
+use std::{iter, mem};
+
+fn usizeify(x: PlainOldULE<8>) -> usize {
+    u64::from_unaligned(&x) as usize
+}
+
+/// A logical representation of the backing `&[u8]` buffer.
+///
+/// This is where the actual work involved in VarZeroVec happens
+///
+/// See [`SliceComponents::new()`] for information on the invariants involved
+pub struct SliceComponents<'a, T> {
+    /// The list of indices into the `things` slice
+    indices: &'a [PlainOldULE<8>],
```

Comment:
I'm fine with making it 32, I'm not really fine with making it smaller. We could make it parametrized with a default I guess

Reply:
I tried with making it a default, a problem is that it requires a fair amount of const generics stuff and while it's possible it would require a lot of additional scaffolding, and furthermore `const N: usize = 4` (defaults) aren't supported for generic parameters.

I'll pick u32 for now.

Reply:
Done